src/graph/UpdateEdgeExecutor.cpp

```diff
@@ -0,0 +1,574 @@
+    int64_t index = -1;
+    for (auto &tagIt : tagProps) {
+        auto status = ectx()->schemaManager()->toTagID(spaceId, tagIt.first);
+        if (!status.ok()) {
+            return Status::Error("No schema found for '%s'", tagIt.first);
+        }
+        auto tagId = status.value();
+        for (auto &prop : tagIt.second) {
+            index++;
+            storage::cpp2::PropDef pd;
+            pd.owner = storage::cpp2::PropOwner::DEST;
+            pd.name = prop;
+            pd.set_tag_id(tagId);
+            props.emplace_back(std::move(pd));
+            dstTagProps_.emplace(std::make_pair(tagIt.first, prop), index);
+        }
+    }
+    return props;
+}
+
+
+StatusOr<std::vector<storage::cpp2::PropDef>> UpdateEdgeExecutor::getReturnProps() {
+    std::vector<storage::cpp2::PropDef> props;
+    auto spaceId = ectx()->rctx()->session()->space();
+    for (auto &tagProp : yieldExpCtx_->srcTagProps()) {
+        storage::cpp2::PropDef pd;
+        pd.owner = storage::cpp2::PropOwner::SOURCE;
+        pd.name = tagProp.second;
+        auto status = ectx()->schemaManager()->toTagID(spaceId, tagProp.first);
+        if (!status.ok()) {
+            return Status::Error("No schema found for '%s'", tagProp.first);
+        }
+        auto tagId = status.value();
+        pd.set_tag_id(tagId);
+        props.emplace_back(std::move(pd));
+        schemaPropIndex_.emplace(std::make_pair(tagProp.first, tagProp.second),
+                                 props.size() - 1);
+    }
+    for (auto &prop : yieldExpCtx_->edgeProps()) {
+        storage::cpp2::PropDef pd;
+        pd.owner = storage::cpp2::PropOwner::EDGE;
+        pd.name = prop;
+        props.emplace_back(std::move(pd));
+        schemaPropIndex_.emplace(std::make_pair(*edgeTypeName_, prop),
+                                 props.size() - 1);
+    }
+    return props;
+}
+
+
+void UpdateEdgeExecutor::insertReverselyEdge() {
+    std::vector<storage::cpp2::Edge> edges;
+    storage::cpp2::Edge reverselyEdge;
+    reverselyEdge.key.set_src(edge_.dst);
+    reverselyEdge.key.set_dst(edge_.src);
+    reverselyEdge.key.set_ranking(edge_.ranking);
+    reverselyEdge.key.set_edge_type(-edgeType_);
+    reverselyEdge.props = "";
+    edges.emplace_back(reverselyEdge);
+    auto space = ectx()->rctx()->session()->space();
+    auto future = ectx()->storage()->addEdges(space, std::move(edges), false);
+    auto *runner = ectx()->rctx()->runner();
+    auto cb = [this] (auto &&resp) {
+        auto completeness = resp.completeness();
+        if (completeness != 100) {
+            // Very bad, it should delete the upsert positive edge!!!
+            DCHECK(onError_);
+            onError_(Status::Error("Internal error about insert reversely edge."));
+            return;
+        }
+    };
+    auto error = [this] (auto &&e) {
+        LOG(ERROR) << "Exception caught: " << e.what();
+        DCHECK(onError_);
+        onError_(Status::Error("Internal error: addEdges"));
```

Comment:
call onError_ in other thread will free "this" pointer, and current thread will get a UAF error.

Reply:
done!